crates/wasi-common/src/snapshots/preview_1.rs

```diff
@@ -1016,36 +1010,34 @@ impl wasi_snapshot_preview1::WasiSnapshotPreview1 for WasiCtx {
         }
     }
 
-    async fn proc_raise(&mut self, _sig: types::Signal) -> Result<(), Error> {
+    async fn proc_raise(&self, _sig: types::Signal) -> Result<(), Error> {
         Err(Error::trap(anyhow::Error::msg("proc_raise unsupported")))
     }
 
-    async fn sched_yield(&mut self) -> Result<(), Error> {
+    async fn sched_yield(&self) -> Result<(), Error> {
         self.sched.sched_yield().await
     }
 
     async fn random_get<'a>(
-        &mut self,
+        &self,
         buf: &GuestPtr<'a, u8>,
         buf_len: types::Size,
     ) -> Result<(), Error> {
         let mut buf = buf
             .as_array(buf_len)
             .as_slice_mut()?
             .expect("cannot use with shared memories; see https://github.com/bytecodealliance/wasmtime/issues/5235 (TODO)");
-        self.random.try_fill_bytes(buf.deref_mut())?;
+        self.random
+            .write()
```

Comment:
I don't think there's any need to serialize multiple threads like this, so the randomness provider should move to a trait with `&self` rather than `&mut self`

Reply:
deferred this to a later PR